psydac/linalg/utilities.py

```diff
@@ -13,46 +14,64 @@
 )
 
 #==============================================================================
-def array_to_psydac(x, Xh):
-    """ converts a numpy array to StencilVector or BlockVector format"""
+def array_to_psydac(x, V):
+    """ 
+    Convert a NumPy array to a Vector of the space V. 
+    This function works in parallel but it is very costly and should be avoided if performance is a priority.
 
-    if isinstance(Xh, BlockVectorSpace):
-        u = BlockVector(Xh)
-        if isinstance(Xh.spaces[0], BlockVectorSpace):
-            for d in range(len(Xh.spaces)):
-                starts = [np.array(V.starts) for V in Xh.spaces[d].spaces]
-                ends   = [np.array(V.ends)   for V in Xh.spaces[d].spaces]
-
-                for i in range(len(starts)):
-                    g = tuple(slice(s,e+1) for s,e in zip(starts[i], ends[i]))
-                    shape = tuple(ends[i]-starts[i]+1)
-                    u[d][i][g] = x[:np.prod(shape) ].reshape(shape)
-                    x          = x[ np.prod(shape):]
+    Parameters
+    ----------
+    x : numpy.ndarray
+        Array to be converted. It only contains the true data, the ghost regions must not be included.
 
-        else:
-            starts = [np.array(V.starts) for V in Xh.spaces]
-            ends   = [np.array(V.ends)   for V in Xh.spaces]
+    V : psydac.linalg.stencil.StencilVectorSpace or psydac.linalg.block.BlockVectorSpace
+        Space where the final Psydac Vector belongs to.
 
-            for i in range(len(starts)):
-                g = tuple(slice(s,e+1) for s,e in zip(starts[i], ends[i]))
-                shape = tuple(ends[i]-starts[i]+1)
-                u[i][g] = x[:np.prod(shape) ].reshape(shape)
-                x       = x[ np.prod(shape):]
+    Returns
+    -------
+    u : psydac.linalg.stencil.StencilVector or psydac.linalg.block.BlockVector
+        Element of space V, the coefficients of which (excluding ghost regions) are the entries of x.
```

Comment:
Maybe mention that the ghost regions of `u` will be up to date?

Reply:
Done in [1bf8f22](https://github.com/pyccel/psydac/pull/404/commits/1bf8f221b454c92dacfafa505bcf03c34ef03827)